core/src/main/java/com/novoda/landingstrip/LandingStripAdapter.java

```diff
@@ -0,0 +1,38 @@
+package com.novoda.landingstrip;
+
+import android.view.View;
+import android.view.ViewGroup;
+
+public abstract class LandingStripAdapter<T extends View> {
+
+    private Listener listener;
+
+    protected abstract T createView(ViewGroup parent, int position);
+
+    protected abstract void bindView(T view, int position);
+
+    void setListener(Listener listener) {
+        this.listener = listener;
+    }
+
+    public void notifyDataSetChanged() {
+        if (listener != null) {
+            listener.onNotifyDataSetChanged(this);
+        }
+    }
+
+    public void notifyItemChanged(int position) {
+        if (listener != null) {
+            listener.onNotifyItemChanged(this, position);
+        }
+    }
+
+    protected abstract int getCount();
+
+    interface Listener {
+
+        <T extends View> void onNotifyDataSetChanged(LandingStripAdapter<T> adapter);
```

Comment:
Why isn't `T` defined as a type parameter on the interface itself? The two methods can currently pass in two different adapters with two different types in theory, it's almost no different than making them pass an adapter of `View`